File: lib/widgets/confirm_destructive.dart
import 'dart:io' show Platform;

import 'package:flutter/cupertino.dart';
import 'package:flutter/foundation.dart' show kIsWeb;
import 'package:flutter/material.dart';

import '../theme/tokens.dart';

/// Ask before something irreversible.
///
/// iOS gets a Cupertino action sheet with a destructive role and Android a
/// Material dialog — HIG and Material both ask for their own alert here rather
/// than a shared custom one. Web falls back to the Material dialog.
Future<bool> confirmDestructive(
  BuildContext context, {
  required String title,
  required String message,
  required String confirmLabel,
  required String cancelLabel,
}) async {
  final isApple = !kIsWeb && (Platform.isIOS || Platform.isMacOS);

  if (isApple) {
    final result = await showCupertinoDialog<bool>(
      context: context,
      barrierDismissible: true,
      builder: (context) => CupertinoAlertDialog(
        title: Text(title),
        content: Text(message),
        actions: [
          CupertinoDialogAction(
            onPressed: () => Navigator.of(context).pop(false),
            child: Text(cancelLabel),
          ),
          CupertinoDialogAction(
            isDestructiveAction: true,
            onPressed: () => Navigator.of(context).pop(true),
            child: Text(confirmLabel),
          ),
        ],
      ),
    );
    return result ?? false;
  }

  final result = await showDialog<bool>(
    context: context,
    builder: (context) => AlertDialog(
      title: Text(title),
      content: Text(message),
      actions: [
        TextButton(
          onPressed: () => Navigator.of(context).pop(false),
          child: Text(cancelLabel),
        ),
        TextButton(
          onPressed: () => Navigator.of(context).pop(true),
          style: TextButton.styleFrom(foregroundColor: AppColor.primary),
          child: Text(confirmLabel),
        ),
      ],
    ),
  );
  return result ?? false;
}

File: src/components/confirmDestructive.ts
import { Alert, Platform } from 'react-native';

/**
 * Ask before something irreversible.
 *
 * On iOS and Android this is the platform's own destructive alert, which is
 * what HIG and Material both call for. `react-native-web` ships `Alert.alert`
 * as a silent no-op, so the web build falls back to the browser's confirm —
 * otherwise the action would appear to do nothing there.
 */
export function confirmDestructive(options: {
  title: string;
  message: string;
  confirmLabel: string;
  cancelLabel: string;
  onConfirm: () => void;
}) {
  const { title, message, confirmLabel, cancelLabel, onConfirm } = options;

  if (Platform.OS === 'web') {
    const ok =
      typeof globalThis !== 'undefined' && typeof globalThis.confirm === 'function'
        ? globalThis.confirm(`${title}\n\n${message}`)
        : false;
    if (ok) onConfirm();
    return;
  }

  Alert.alert(
    title,
    message,
    [
      { text: cancelLabel, style: 'cancel' },
      { text: confirmLabel, style: 'destructive', onPress: onConfirm },
    ],
    { cancelable: true },
  );
}
